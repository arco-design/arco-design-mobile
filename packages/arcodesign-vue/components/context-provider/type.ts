export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
