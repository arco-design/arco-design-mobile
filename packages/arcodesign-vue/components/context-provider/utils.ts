import { InjectionKey, inject } from 'vue';
import { ContextInstallOptions, GlobalContextParams } from './type';

export const defaultInstallOptions: Required<ContextInstallOptions> = {
    componentPrefix: 'Arco',
};

export const defaultContext: Required<GlobalContextParams> = {
    prefixCls: 'arco',
};

export const configProviderInjectionKey: InjectionKey<GlobalContextParams> =
    Symbol('ArcoContextProvider');

export function getComponentPrefix(options?: ContextInstallOptions) {
    return options?.componentPrefix || defaultInstallOptions.componentPrefix;
}

export function getPrefixCls(componentName?: string) {
    const contextProvider = inject(configProviderInjectionKey, defaultContext);
    const prefix = contextProvider.prefixCls;
    if (componentName) {
        return `${prefix}-${componentName}`;
    }
    return prefix;
}
