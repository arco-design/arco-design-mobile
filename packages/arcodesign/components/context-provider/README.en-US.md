### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
