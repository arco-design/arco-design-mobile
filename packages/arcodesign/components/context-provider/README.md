### 其他

# 全局配置 ContextProvider

全局数据控制组件，用于替换全局数据。

======

> 属性/Props

|参数|描述|类型|默认值|
|----------|-------------|------|------|
|prefixCls|组件类名前缀|string|"arco"|
|system|手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|是否使用暗黑模式|boolean|false|
|isDarkMode|是否处于暗黑模式，指定后以指定的值为准|boolean|false|
|theme|主题变量，传入后将在线替换css变量，需设置less变量 @use\-css\-vars: 1|Record\<string, string\>|-|
|locale|国际化语言包配置|ILocale|-|
|useRtl|是否使用Rtl模式|boolean|false|

> ILocale

|参数|描述|类型|
|----------|-------------|------|
|locale|语言类型|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
