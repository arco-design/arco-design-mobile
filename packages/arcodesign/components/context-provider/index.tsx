import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
