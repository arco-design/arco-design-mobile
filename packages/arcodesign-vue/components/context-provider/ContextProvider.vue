<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
