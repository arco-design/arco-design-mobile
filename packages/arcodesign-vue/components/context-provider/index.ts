import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
