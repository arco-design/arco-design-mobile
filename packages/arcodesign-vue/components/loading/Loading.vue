<script lang="ts" setup>
import { getPrefixCls } from '../context-provider';
</script>

<template>
    <div></div>
</template>
