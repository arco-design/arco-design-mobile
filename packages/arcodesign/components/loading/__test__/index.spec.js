import React from 'react';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import Loading from '..';
import { mount } from 'enzyme';
import { defaultContext } from '../../context-provider';

const prefix = `${defaultContext.prefixCls}-loading`;

demoTest('loading');

mountTest(Loading, 'Loading');

describe('Loading', () => {
    it('Different types render correctly', () => {
        const wrapper = mount(
            <div>
                <Loading type="arc" />
                <Loading type="circle" />
                <Loading type="spin" />
                <Loading type="dot" />
            </div>
        );
        expect(wrapper.find(`.${prefix}.all-border-box.arc svg`).length).toBe(1);
        expect(wrapper.find(`.${prefix}.all-border-box.circle svg`).length).toBe(1);
        expect(wrapper.find(`.${prefix}.all-border-box.spin span`).length).toBe(16);
        expect(wrapper.find(`.${prefix}.all-border-box.dot span`).length).toBe(3);
    });
})
