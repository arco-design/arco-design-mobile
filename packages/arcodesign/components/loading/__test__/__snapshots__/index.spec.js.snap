// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading demo test loading demo: custom-color.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loading-demo-basic"
  >
    <div
      class="arco-loading all-border-box arc"
      style="animation-duration: 1000ms; width: 20px; height: 20px;"
    >
      <svg
        viewBox="0 0 20 20"
      >
        <circle
          class="arc-bg"
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke-width="2"
        />
        <circle
          class="arc-line stroke-color-with-config"
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke-dasharray="14.137166941154069 42.411500823462205"
          stroke-dashoffset="14.137166941154069"
          stroke-linecap="round"
          stroke-width="2"
          style="stroke: #ff5722;"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box circle"
      style="animation-duration: 1000ms; width: 20px; height: 20px;"
    >
      <svg
        viewBox="0 0 20 20"
      >
        <defs>
          <lineargradient
            id="grad1-inner-1"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
              style="stop-color: rgb(255, 87, 34);"
            />
            <stop
              class="loading-circle-start stop-color-with-config"
              offset="100%"
              style="stop-color: rgb(255, 87, 34);"
            />
          </lineargradient>
          <lineargradient
            id="grad2-inner-1"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
              style="stop-color: rgb(255, 87, 34);"
            />
            <stop
              class="loading-circle-end stop-color-with-config"
              offset="100%"
              style="stop-color: rgb(255, 87, 34);"
            />
          </lineargradient>
        </defs>
        <circle
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke="url(#grad1-inner-1)"
          stroke-dasharray="28.274333882308138"
          stroke-dashoffset="28.274333882308138"
          stroke-width="2"
        />
        <circle
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke="url(#grad2-inner-1)"
          stroke-dasharray="28.274333882308138"
          stroke-width="2"
        />
        <circle
          class="loading-circle-filleted fill-color-with-config"
          cx="19"
          cy="10"
          r="1"
          style="fill: #ff5722;"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box spin"
      style="animation-duration: 1000ms;"
    >
      <span
        class="spin-cell"
        style="opacity: 1; transform: rotate(0turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0; transform: rotate(0.125turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.1; transform: rotate(0.25turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.25; transform: rotate(0.375turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.4; transform: rotate(0.5turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.55; transform: rotate(0.625turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.7; transform: rotate(0.75turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.85; transform: rotate(0.875turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="background-color: rgb(255, 87, 34); border-radius: 2px;"
        />
      </span>
    </div>
    <div
      class="arco-loading all-border-box dot"
      style="animation-duration: 1000ms;"
    >
      <span
        class="dot-cell bg-color-with-config filleted"
        style="opacity: 0.1; background-color: rgb(96, 106, 120);"
      />
      <span
        class="dot-cell bg-color-with-config filleted"
        style="opacity: 0.3; background-color: rgb(96, 106, 120);"
      />
      <span
        class="dot-cell bg-color-with-config filleted"
        style="opacity: 0.5; background-color: rgb(96, 106, 120);"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`loading demo test loading demo: custom-size.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loading-demo-basic"
  >
    <div
      class="arco-loading all-border-box arc"
      style="animation-duration: 1000ms; width: 16px; height: 16px;"
    >
      <svg
        viewBox="0 0 16 16"
      >
        <circle
          class="arc-bg"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-width="2"
        />
        <circle
          class="arc-line stroke-color-with-config"
          cx="8"
          cy="8"
          fill="none"
          r="7"
          stroke-dasharray="10.995574287564276 32.98672286269283"
          stroke-dashoffset="10.995574287564276"
          stroke-linecap="round"
          stroke-width="2"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box circle"
      style="animation-duration: 1000ms; width: 31px; height: 31px;"
    >
      <svg
        viewBox="0 0 31 31"
      >
        <defs>
          <lineargradient
            id="grad1-inner-5"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
            />
            <stop
              class="loading-circle-start stop-color-with-config"
              offset="100%"
            />
          </lineargradient>
          <lineargradient
            id="grad2-inner-5"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
            />
            <stop
              class="loading-circle-end stop-color-with-config"
              offset="100%"
            />
          </lineargradient>
        </defs>
        <circle
          cx="15.5"
          cy="15.5"
          fill="none"
          r="15"
          stroke="url(#grad1-inner-5)"
          stroke-dasharray="47.12388980384689"
          stroke-dashoffset="47.12388980384689"
          stroke-width="1"
        />
        <circle
          cx="15.5"
          cy="15.5"
          fill="none"
          r="15"
          stroke="url(#grad2-inner-5)"
          stroke-dasharray="47.12388980384689"
          stroke-width="1"
        />
        <circle
          class="loading-circle-filleted fill-color-with-config"
          cx="30.5"
          cy="15.5"
          r="0.5"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box spin"
      style="animation-duration: 1000ms;"
    >
      <span
        class="spin-cell"
        style="opacity: 1; transform: rotate(0turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.1; transform: rotate(0.125turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.2286; transform: rotate(0.25turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.3572; transform: rotate(0.375turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.4858; transform: rotate(0.5turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.6144; transform: rotate(0.625turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.743; transform: rotate(0.75turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.8716; transform: rotate(0.875turn); width: 1px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 1px;"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`loading demo test loading demo: filleted.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loading-demo-basic"
  >
    <div
      class="arco-loading all-border-box arc"
      style="animation-duration: 1000ms; width: 20px; height: 20px;"
    >
      <svg
        viewBox="0 0 20 20"
      >
        <circle
          class="arc-bg"
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke-width="2"
        />
        <circle
          class="arc-line stroke-color-with-config"
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke-dasharray="14.137166941154069 42.411500823462205"
          stroke-dashoffset="14.137166941154069"
          stroke-width="2"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box circle"
      style="animation-duration: 1000ms; width: 20px; height: 20px;"
    >
      <svg
        viewBox="0 0 20 20"
      >
        <defs>
          <lineargradient
            id="grad1-inner-8"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
            />
            <stop
              class="loading-circle-start stop-color-with-config"
              offset="100%"
            />
          </lineargradient>
          <lineargradient
            id="grad2-inner-8"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
            />
            <stop
              class="loading-circle-end stop-color-with-config"
              offset="100%"
            />
          </lineargradient>
        </defs>
        <circle
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke="url(#grad1-inner-8)"
          stroke-dasharray="28.274333882308138"
          stroke-dashoffset="28.274333882308138"
          stroke-width="2"
        />
        <circle
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke="url(#grad2-inner-8)"
          stroke-dasharray="28.274333882308138"
          stroke-width="2"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box spin"
      style="animation-duration: 1000ms;"
    >
      <span
        class="spin-cell"
        style="opacity: 1; transform: rotate(0turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.1; transform: rotate(0.125turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.2286; transform: rotate(0.25turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.3572; transform: rotate(0.375turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.4858; transform: rotate(0.5turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.6144; transform: rotate(0.625turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.743; transform: rotate(0.75turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.8716; transform: rotate(0.875turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
        />
      </span>
    </div>
    <div
      class="arco-loading all-border-box dot"
      style="animation-duration: 1000ms;"
    >
      <span
        class="dot-cell bg-color-with-config"
        style="opacity: 0.2;"
      />
      <span
        class="dot-cell bg-color-with-config"
        style="opacity: 0.6;"
      />
      <span
        class="dot-cell bg-color-with-config"
        style="opacity: 1;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`loading demo test loading demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loading-demo-basic"
  >
    <div
      class="arco-loading all-border-box arc"
      style="animation-duration: 1000ms; width: 20px; height: 20px;"
    >
      <svg
        viewBox="0 0 20 20"
      >
        <circle
          class="arc-bg"
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke-width="2"
        />
        <circle
          class="arc-line stroke-color-with-config"
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke-dasharray="14.137166941154069 42.411500823462205"
          stroke-dashoffset="14.137166941154069"
          stroke-linecap="round"
          stroke-width="2"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box circle"
      style="animation-duration: 1000ms; width: 20px; height: 20px;"
    >
      <svg
        viewBox="0 0 20 20"
      >
        <defs>
          <lineargradient
            id="grad1-inner-12"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
            />
            <stop
              class="loading-circle-start stop-color-with-config"
              offset="100%"
            />
          </lineargradient>
          <lineargradient
            id="grad2-inner-12"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="0%"
          >
            <stop
              class="loading-circle-middle stop-color-with-config"
              offset="0%"
            />
            <stop
              class="loading-circle-end stop-color-with-config"
              offset="100%"
            />
          </lineargradient>
        </defs>
        <circle
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke="url(#grad1-inner-12)"
          stroke-dasharray="28.274333882308138"
          stroke-dashoffset="28.274333882308138"
          stroke-width="2"
        />
        <circle
          cx="10"
          cy="10"
          fill="none"
          r="9"
          stroke="url(#grad2-inner-12)"
          stroke-dasharray="28.274333882308138"
          stroke-width="2"
        />
        <circle
          class="loading-circle-filleted fill-color-with-config"
          cx="19"
          cy="10"
          r="1"
        />
      </svg>
    </div>
    <div
      class="arco-loading all-border-box spin"
      style="animation-duration: 1000ms;"
    >
      <span
        class="spin-cell"
        style="opacity: 1; transform: rotate(0turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.1; transform: rotate(0.125turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.2286; transform: rotate(0.25turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.3572; transform: rotate(0.375turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.4858; transform: rotate(0.5turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.6144; transform: rotate(0.625turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.743; transform: rotate(0.75turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
      <span
        class="spin-cell"
        style="opacity: 0.8716; transform: rotate(0.875turn); width: 2px;"
      >
        <span
          class="spin-cell-inner bg-color-with-config"
          style="border-radius: 2px;"
        />
      </span>
    </div>
    <div
      class="arco-loading all-border-box dot"
      style="animation-duration: 1000ms;"
    >
      <span
        class="dot-cell bg-color-with-config filleted"
        style="opacity: 0.2;"
      />
      <span
        class="dot-cell bg-color-with-config filleted"
        style="opacity: 0.6;"
      />
      <span
        class="dot-cell bg-color-with-config filleted"
        style="opacity: 1;"
      />
    </div>
  </div>
</DocumentFragment>
`;
